### PR TITLE
Localice cache changes

### PR DESCRIPTION
**Pull request by ajtret**

### CODE REVIEW

File: src/integrations/clevertap/client/ClevertapApiClient.ts [hunk 1]
```diff
@@ -120,6 +123,26 @@ export class ClevertapApiClient extends APIAbstract {
     })
   }
 
+  public async getContentBlocks(
```
Comment (parastaneja2709) [inline src/integrations/clevertap/client/ClevertapApiClient.ts:126]: nit: can we rename it to getContentBlockTemplates?

File: src/integrations/clevertap/client/clevertapApiTypes.ts [hunk 2]
```diff
@@ -25,6 +26,13 @@ export type ClevertapTemplatesList = {
   pageSize: number
 }
 
+export type ClevertapContentBlockList = {
+  contentBlock: ClevertapContentBlockItem[]
```
[the remark this answers is not in the document]
Comment (parastaneja2709) [reply src/integrations/clevertap/client/clevertapApiTypes.ts:30]: Here we can use the union type like this:
export type ClevertapTemplatesList = {
  total: number;
  pageNumber: number;
  pageSize: number;
} & (
  | { templates: ClevertapTemplatesListItem[]; contentBlock?: never }
  | { contentBlock: ClevertapContentBlockItem[]; templates?: never }
);
or we can use the generics

File: src/modules/cache/CacheService.ts [hunk 3]
```diff
@@ -33,4 +33,21 @@ export class CacheService extends DynamicHostService {
       ids,
     )
   }
+
```
Comment (parastaneja2709) [inline src/modules/cache/CacheService.ts:36]: This is not required. Content block templates will be fetched using listItems and getItems request handler only

File: src/integrations/clevertap/client/clevertapApiTypes.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ajtret) [reply src/integrations/clevertap/client/clevertapApiTypes.ts]: Created different types for content block

File: src/modules/publish/publishSchemas.ts [hunk 4]
```diff
@@ -21,4 +21,18 @@ export const publishResponseBody = z.object({
...
 })
 
+export const cmsPublishResponseBody = z.object({
+  statusCode: z.number(),
+  message: z.string().optional(),
+  payload: z
+    .object({
+      html: z.string(),
+      repalcements: z.string(),
+      text: z.object({
```
[the remark this answers is not in the document]
Comment (ajtret) [reply src/modules/publish/publishSchemas.ts]: done

File: src/integrations/clevertap/client/ClevertapApiClient.ts [hunk 5]
```diff
@@ -1,24 +1,24 @@
...
+import type {
+  LocaleDefinition,
+  SuccessMessageResponse,
+  UpdateEmailTemplateRequestBody,
+  UpdateTemplateParams,
+} from '../types'
 
 import { APIAbstract } from './APIAbstract'
-import {
+import type {
```
Comment (unmesh46) [inline src/integrations/clevertap/client/ClevertapApiClient.ts:14]: remove type

File: src/integrations/clevertap/client/ClevertapApiClient.ts [hunk 6]
```diff
@@ -148,6 +172,31 @@ export class ClevertapApiClient extends APIAbstract {
...
+        'X-CleverTap-Account-Id': accountId,
+        'X-CleverTap-Passcode': passcode,
+        'content-type': 'application/json',
+      },
+      query: {
+        id: templateId,
+      },
+      requestLabel: 'getEmailTemplateById',
+    })
+    // const aj: ClevertapEmailTemplate = {
```
Comment (unmesh46) [inline src/integrations/clevertap/client/ClevertapApiClient.ts:187]: Remove comments

File: src/integrations/clevertap/client/clevertapApiTypes.ts [hunk 7]
```diff
@@ -66,14 +98,38 @@ export type ClevertapEmailTemplate = {
   updatedBy?: string
   partner?: string
   locale?: string
+  id?: number
+  name?: string
+  content?: string
+  type?: string
 }
 
+// export type ClevertapContentBlock = {
```
Comment (unmesh46) [inline src/integrations/clevertap/client/clevertapApiTypes.ts:107]: Remove comment

File: src/modules/publish/publishSchemas.ts [hunk 8]
```diff
@@ -21,4 +21,18 @@ export const publishResponseBody = z.object({
     .optional(),
 })
 
+export const cmsPublishResponseBody = z.object({
```
Comment (unmesh46) [inline src/modules/publish/publishSchemas.ts:24]: Not used anywhere, remove this method.

**Pull request merged**